adal/oauth2_client.py

```diff
@@ -305,7 +305,10 @@ def get_user_code_info(self, oauth_parameters):
             raise
 
         if util.is_http_success(resp.status_code):
-            return self._handle_get_device_code_response(resp.text)
+            user_code_info = self._handle_get_device_code_response(resp.text)
+            if resp and resp.headers and resp.headers.get('client-request-id'):

```

Comment:
Such defensive programming style is generally a good thing. But in this case it is unnecessary. Because resp already has `headers` as one of its [officially documented property](http://docs.python-requests.org/en/master/api/#requests.Response). You can simply do `user_code_info['correlation_id'] = resp.headers.get('client-request-id')`.

Reply:
I was following the convention set in util.log_return_correlation_id. Should we remove those too?

Reply:
Good question! And the answer is no. Inside the scope of `log_return_correlation_id()`, the response is an input parameter that we don't exactly know what it is (remember that Python, as many scripting language, does not have static typing). So defensive programming is desirable. But here, we already know exactly what resp is, based on line 297; also the dominant convention of no-need-to-be-defensive is set up at line 307 already.

Besides, have you addressed my 2nd half of [previous comment for line 115](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/130#discussion_r175137770)?

Reply:
Great, that makes sense. I will make the fix only in oauth2_client